console/console-init/ui/src/modules/device-detail/components/DeviceInfo/DeviceInfo.tsx

```diff
@@ -46,23 +58,35 @@ export const DeviceInfo: React.FC<IDeviceInfoProps> = ({
             <SwitchWithToggle
               id={"divice-info-view-json"}
               label={"View in JSON"}
+              onChange={onToggle}
             />
           </SplitItem>
         </Split>
         <br />
-        <Grid>
-          <GridItem span={5} className={styles.gateways_align}>
-            <DeviceInfoGateways deviceList={deviceList} />
-            <br />
-            <DeviceInfoCredentials />
-          </GridItem>
-          <GridItem span={7}>
-            <DeviceInfoMetadata
-              dataList={metadataList}
-              id={"divice-info-metadata"}
-            />
-          </GridItem>
-        </Grid>
+        {isHidden ? (
+          <JsonViewEditor
+            readOnly={true}
+            detailInJson={jsonViewData}
```

Comment:
Please revisit the naming of the prop detailInJson.

Reply:
This parameter defined in generic component JsonViewEditor. If I'll amke change, I will have to make change everywhere.
Sanjeev  is doing refactoring on this component. We can suggest the on sanjeev's PR.